app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -258,12 +262,29 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
     )
   }
 
-  fun markHomeMenuCloseDrawer() {
+  fun markLastCheckedItemCloseDrawer(lastCheckedItemId: Int, isAdminSelected: Boolean) {
+    if (isAdminSelected) {
+      getFooterViewModel().isAdministratorControlsSelected.set(true)
+    } else if (lastCheckedItemId != -1) {
+      binding.fragmentDrawerNavView.menu.getItem(
+        when (lastCheckedItemId) {
```

Comment:
keep when outside of getItem() and use a var to input in getItem(), this way it will be more clean and readable.

Reply:
Done